src/IssueBookDao.java

```diff
@@ -0,0 +1,63 @@
+import java.sql.*;
+public class IssueBookDao {
+	
+public static boolean checkBook(String bookcallno){
+	boolean status=false;
+	try{
+		Connection con=DB.getConnection();
+		PreparedStatement ps=con.prepareStatement("select * from books where callno=?");
+		ps.setString(1,bookcallno);
+	    ResultSet rs=ps.executeQuery();
+		status=rs.next();
+		con.close();
+	}catch(Exception e){System.out.println(e);}
+	return status;
+}
+
+public static int save(String bookcallno,int studentid,String studentname,String studentcontact){
+	int status=0;
+	try{
+		Connection con=DB.getConnection();
+		
+		status=updatebook(bookcallno);//updating quantity and issue
+		
+		if(status>0){
+		PreparedStatement ps=con.prepareStatement("insert into issuebooks(bookcallno,studentid,studentname,studentcontact) values(?,?,?,?)");

```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer

**Problem**
This line of code contains a resource that might not be closed properly. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *ps*. The resource is referenced in statements at the following lines: 25, 26, 27, 28, 29, 30. The resource closure statement is at line: 33. There are other execution paths that do not contain closure statements, e.g., when exception is thrown by PreparedStatement.executeUpdate. Either a) close *ps* in a try-finally block or b) close the resource by declaring *ps* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).